src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -413,8 +414,9 @@ QgsPostgresConn::QgsPostgresConn( const QString &conninfo, bool readOnly, bool s
 
   if ( mPostgresqlVersion >= 90000 )
   {
-    LoggedPQexecNR( "QgsPostgresConn", QStringLiteral( "SET application_name='QGIS'" ) );
-    LoggedPQexecNR( "QgsPostgresConn", QStringLiteral( "SET extra_float_digits=3" ) );
+    QgsSettings settings;
+    const QString applicationName = settings.value( QStringLiteral( "PostgreSQL/application_name" ), PG_APPLICATION_NAME, QgsSettings::Providers ).toString();
+    LoggedPQexecNR( "QgsPostgresConn", QStringLiteral( "SET application_name=%s; SET extra_float_digits=3" ).arg( quotedValue( applicationName ) ) );
```

Comment:
`SET extra_float_digits=3`
is this related to settings the application name? 
What it does?

Reply:
Besides, shouldn't be better to default to  QCoreApplication::applicationName() plus QgsApplication::platform() ?

Reply:
The `SET extra_float_digits` was already there before this PR. It was a suggestion from @MorriganR to combine these 2 queries together https://github.com/qgis/QGIS/pull/49468#discussion_r929671570

Using `QgsApplication::platform()` might be a good idea for the default value indeed.
For the PR, they are several QGIS server running, we would like to be able to distinguish them.

Reply:
Hi @elpaso thanks for the review.

The SET extra_float_digits was already there before this PR. It was a suggestion from @MorriganR to combine these 2 queries together https://github.com/qgis/QGIS/pull/49468#discussion_r929671570

Using `QgsApplication::platform()` for default application name is a good idea. I have do it.

Reply:
You can also use  QCoreApplication::applicationName() to get "QGIS" (through QgsApplication::QGIS_APPLICATION_NAME) , there is no need for another static variable.

But I have another concern here:

if this is an option to be used in QGIS Desktop you probably need to expose it to the GUI.
if this is an option to be used in QGIS Server you need to expose it as a new QGIS_SERVER_.... env variable config option and document it.

Generally speaking, I understand your use case, I'm only doubtful about how widely used this option will be (especially if it was designed for the server context where our user base is rather narrow), I am just worried about adding another features to QGIS that nobody (except the authors) will use.

Note that I might be completely wrong here and this is one of the most wanted features.

Reply:
I have check `QCoreApplication::applicationName() ` and it returns `QGIS3` not only `QGIS`. https://github.com/qgis/QGIS/blob/0c577f07ba45e3903eb09b5239d2d5da572e8f6c/src/core/qgsapplication.cpp#L152

Then I have made `application_name` an option in QGIS3.conf file like the PostgreSQL connection time_out. You don't have any GUI or any env varaible for connection timeout config. I was thinking that the application_name is much more something to configure at the install. For me it depends on organization and way to install QGIS or QGIS Server.

@elpaso do you know where the PostgreSQL provider can be configure ?

Reply:
> I have check `QCoreApplication::applicationName() ` and it returns `QGIS3` not only `QGIS`.

Is that a problem?

> 
> https://github.com/qgis/QGIS/blob/0c577f07ba45e3903eb09b5239d2d5da572e8f6c/src/core/qgsapplication.cpp#L152
> 
> Then I have made `application_name` an option in QGIS3.conf file like the PostgreSQL connection time_out. You don't have any GUI or any env varaible for connection timeout config. I was thinking that the application_name is much more something to configure at the install. For me it depends on organization and way to install QGIS or QGIS Server.

In general we should follow the 12 factors paradigm (https://12factor.net/) which mandates config through the environment.

We should definitely think of a generic way to override QGIS3.ini settings through the env if we want to become serious about web map services deployment on complex infrastructures.

> 
> @elpaso do you know where the PostgreSQL provider can be configure ?

There isn't any except per-connection settings. IMHO this is a legacy of the server being created on top of the QGIS desktop application.

Reply:
> The `SET extra_float_digits` was already there before this PR. It was a suggestion from @MorriganR to combine these 2 queries together [#49468 (comment)](https://github.com/qgis/QGIS/pull/49468#discussion_r929671570)
> 

Fair enough, a comment in the code wouldn't hurt.